java/arcs/android/util/AndroidBinderStats.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.util
+
+import android.os.Process
+import arcs.core.util.TaggedLog
+import java.io.File
+
+/**
+ * An utility singleton queries binder stats directly from the kernel binder driver.
+ *
+ * Note:
+ * If a permission denial shows up at i.e. a privileged application, try to execute the command
+ * to configure secure Linux to permissive mode:
+ * $> adb shell setenforce 0
+ *
+ * By default, a privileged application is disallowed to access any debugfs file attributes.
+ * neverallow priv_app debugfs:file read;
+ */
+object AndroidBinderStats {
+    private const val STATS_FILE_NODE = "/sys/kernel/debug/binder/stats"
+    private const val PROCESS_TAG = "proc "
+    private val log = TaggedLog { "AndroidBinderStats" }
+
+    /** Query the stats of the given binder process record [tags]. */
+    fun query(vararg tags: String): List<String> =
+        with(parse()) {
+            tags.map { this[it] ?: "" }
+        }
+
+    private fun parse(): Map<String, String> {
+        try {
+            File(STATS_FILE_NODE).useLines { lines ->
+                val delimiter = PROCESS_TAG + Process.myPid()
+                var partitionFlip = false
+                return lines.partition {
```

Comment:
Rather than returning for the `parse` function here, try this:

```kotlin
private fun parse(): Map<String, String> {
    return try {
        File(STATS_FILE_NODE).useLines {
            ...
            lines.partition {
                ...
            }.first.associate {
                ...
            }
        }
    } catch (e: Exception) {
        log.info { e.message }
        emptyMap()
    }
}
```

Reply:
Hmm, ok, use try/catch as an expression. Thanks for the suggestion, will change.